src/Uno.UI/UI/Xaml/Controls/MenuFlyout/CascadingMenuHelper.cs

```diff
@@ -0,0 +1,977 @@
+			UIElement ownerAsUIE = m_wpOwner?.Target as UIElement;
+
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnVisibilityChanged - ownerAsUIE=0x%p.", this, ownerAsUIE));
+#endif // CMH_DEBUG
+
+			if (ownerAsUIE != null)
+			{
+				Visibility visibility = Visibility.Collapsed;
+				visibility = ownerAsUIE.Visibility;
+
+				if (Visibility.Visible != visibility)
+				{
+					ClearStateFlags();
+				}
+			}
+		}
+
+		internal void OnPresenterSizeChanged(
+			object pSender,
+			SizeChangedEventArgs args,
+			Controls.Popup popup)
+		{
+			Control ownerAsControl = m_wpOwner?.Target as Control;
+
+			ISubMenuOwner ownerAsSubMenuOwner = m_wpOwner?.Target as ISubMenuOwner;
+
+			Control presenterAsControl = m_wpSubMenuPresenter?.Target as Control;
+
+#if CMH_DEBUG
+			(DebugTrace(XCP_TRACE_OUTPUT_MSG /*traceType*/, "CMH[0x%p]: OnPresenterSizeChanged - ownerAsControl=0x%p, ownerAsSubMenuOwner=0x%p.", this, ownerAsControl, ownerAsSubMenuOwner));
+#endif // CMH_DEBUG
+
+			if (ownerAsControl != null && ownerAsSubMenuOwner != null)
+			{
+				Size newPresenterSize = default;
+				newPresenterSize = args.NewSize;
+
+				FlowDirection flowDirection = FlowDirection.LeftToRight;
+				flowDirection = (ownerAsControl.FlowDirection);
+
+				// We sometimes will only want to change one of the two XY-positions of the menu,
+				// but some menus (e.g. AppBarButton.Flyout) don't allow you to individually change
+				// one axis of the position - you need to close and reopen the menu in a different location.
+				// This necessitates a single function call that takes a Point parameter rather than
+				// two function calls that individually change the X and then the Y position of the menu,
+				// since otherwise we'd be closing and reopening the menu twice if we needed to change
+				// both positions, which would be visually disruptive.
+				// As such, we need a way to tell PositionSubMenu to leave one of the positions as it was.
+				// We'll use negative infinity as a sentinel value that means "don't change this coordinate value".
+				Point targetPoint = new Point(float.NegativeInfinity, float.NegativeInfinity);
+
+				bool isPositionedAbsolutely = false;
+				isPositionedAbsolutely = (ownerAsSubMenuOwner.IsSubMenuPositionedAbsolutely);
+
+				Point positionPoint = new Point(0, 0);
+
+				if (isPositionedAbsolutely)
+				{
+					// Get the current sub menu item target position as the client point
+					GeneralTransform transformToRoot;
+					transformToRoot = ownerAsControl.TransformToVisual(null);
+					positionPoint = transformToRoot.TransformPoint(positionPoint);
+				}
+
+				// Get the current sub menu item width and height
+				double width = 0;
+				double height = 0;
+				width = (ownerAsControl.ActualWidth);
+				height = (ownerAsControl.ActualHeight);
+
+				// Get the current presenter max width/height
+				double maxWidth = Double.NaN;
+				double maxHeight = Double.NaN;
+				maxWidth = (presenterAsControl.MaxWidth);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['presenterAsControl' is null on at least one execution path.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)